### ISSUE TITLE
Details on Inv Depth Layer and ConstantPad2D

### ISSUE THREAD

**Issue by F-Barto** [TRI-ML/packnet-sfm#247]
Hello,

I would like to know why are you explicitly using ConstantPad2D ?

[[link to code]](https://github.com/TRI-ML/packnet-sfm/blob/de53b310533ff6b01eaa23a8ba5ac01bac5587b1/packnet_sfm/networks/layers/packnet/layers01.py#L116)
```
self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1)
self.pad = nn.ConstantPad2d([1] * 4, value=0)
```

From the [doc ](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html?highlight=conv2d#torch.nn.Conv2d) nn.Conv2d default padding_mode is to `zeros`. Isn't setting `padding=1` in Conv2d equivalent to your self.conv1(self.pad(x)) ?

Am I missing something ?

Have a good day
